### ISSUE TITLE
Build persons module(s)

### ISSUE THREAD

**Issue by wlpotter** [wlpotter/csv-to-srophe#5]

**Comment by wlpotter** [comment]
Second draft of the updated persons input data table is here: https://docs.google.com/spreadsheets/d/1tlEAM3rn_Yk2Dkkxj81czoqgv0Viec16v-q-ubisluY/edit#gid=816362317 (see the revision1 sheet). The original column names are kept in row 2 for reference. Here some remaining questions to be addressed before I begin implementing:

1. Columns B and C: These are blank in the current data; not sure what they may be used for, of if and how the script should handle them.
2. Columns G and H (sourceUri.headword.syr; citedRange.headword.syr): what are these citations for? Just the Syriac name, the English name, or both?
3. Columns O, P, Q and R (and S) (abstract.en; anonymousDesc.en; sourceUri.abstract.en; citedRange.abstract.en; citationUnit.abstract.en): Columns O and P, I know, are still in flux, but what are Q and R sources for? If we end up having a 5-words-or-less anonymous description, it seems like we would want that to be '@resp="http://syriaca.org' whereas the abstract (column O) would be sourced. Compare, for instance, Ephrem (http://syriaca.org/person/13). Of course, if Q and R are blank, we would have the abstract marked with a @resp attribute as well.
    - if we would want a place to provide a source for the anonymous description, we can add the sourceUri, citedRange, and citationUnit columns for column P (Q, R, and S would then be for column O).
4. Columns T-X (date1; when.date1; notBefore.date1; notAfter.date1; type.date1), and Y-AA (sourceUri.date1; citedRange.date1; citationUnit.date1): I switched to the new dating system. I kept the enumeration ("date1") to allow for another application of the transform to parse data with multiple dates. By convention, "date1" contains the text node of the output element; when., notBefore., and notAfter.date1 all contain the machine-readable attribute values (I will have the script convert them to four-digit dates with leading zeroes as needed but BCE will need to have the negative sign); type.date1 contains the string "floruit", "death", or "birth" which will determine how the data is encoded. The sourceUri, etc. columns now refer to the date regardless of the date type.
    - for a few of these, I added text to the date1 column. The text node can also just be created via the attribute values. The only exception here that I found is row 121 (Anonymous 3755) which has a @notBefore value of 0520, which might be hard to generate just from the attribute values. I'm not sure if we should enforce some kind of text in the date1 column or have the script handle blank cells as best it can?
5. Columns AD-AE (sourceUri.name1; citedRange.name1; citationUnit.name1): In other source columns I added a citationUnit.x (see, e.g., AA, citationUnit.date1). This is intended to solve the same problem as Columns AD-AE solve where you have multiple cited ranges with different units (pages, section, entry, etc.). AD-AE illustrate an issue to my approach when you have more than one citation unit in a given citation. I think there are two main options:
    - have columns for each needed citation unit (this is how the previous sheet solved this). So you would have citedRange.pp.name1, citedRange.section.name1, etc. The benefit is that you don't have to worry about entering the citation type as long as you put the range in the correct column. The downside is an exponential growth of columns as each cited range may need 3 or more extra citedRange columns.
    - allow a sequence of character-separated values in the citedRange.x column and the citationUnit.x column. This is similar to what we talked about for nested entities in text nodes, but I don't think it will be as messy. For each source you'd still only have the sourceUri.x, citedRange.x, and citationUnit.x columns. However, the latter two can hold multiple values. So, for instance, you could have "333#Abraham" in citedRange.x and "pp#entry" in citationUnit.x. See row 32 for my implementation of this. I've used the "#" character as the pipe character ("|") can sometimes be used as the separator in a csv file.

**Comment by wlpotter** [comment]
Further questions on this data:

1. Will any of the persons in the initial spreadsheet be authors and/or saints? If so, we should add a column for that data (as it will trigger an `@ana` attribute on the tei:person element and will change how the tei:seriesStmt element(s) look.

**Comment by dlschwartz** [comment]
@wlpotter This is a possibility. I actually need to discuss this with Dave and Jeanne-Nicole. From the perspective of SPEAR, anyone a text calls a saint is a saint. I'm not sure what the definition is for Syriaca more broadly.

Another related question is how we handle describing someone as "anonymous". It's currently with the use of tei:label. However, I'm wondering if we don't want to make this the equivalent of "saint" and "author." We might want to display (or not display) these people as a class like the others. Winona can make that work with tei:label but maybe these should all be the same. Perhaps we need to discuss this before moving forward.

**Comment by wlpotter** [comment]
@dlschwartz this is also a good question. I raised an issue for treating the `srophe:tag` values in the same way (https://github.com/srophe/srophe-app-data/issues/914), but it's worth discussing the separate issue of how those values should be interpreted by the app and by users.

I also noticed in connection with this question that we have on 1019 records (based on a simple find-in-files) that have a `/person/@xml:id` value of "saint-\d+" (where \d+ is the URI of the record) as opposed to the more common "person-\d+". There are no examples of "author-\d+". If we do have saints records coming from the spreadsheet, will we want to maintain this convention? The point may be moot as I think the xml:ids on the person element will be removed with the schema and batch updates, which would mirror what we did with places.

Adding all this here as a reminder to discuss at some point.
